eth/core/state/statedb.go

```diff
@@ -177,14 +181,19 @@ func (sdb *stateDB) SlotInAccessList(addr common.Address, slot common.Hash) (boo
 	return sdb.Contains(addr, slot)
 }
 
-// TODO: `GetTransientState` implements the `PolarisStateDB` interface by returning the transient state
+// `GetTransientState` implements `stateDB`
 func (sdb *stateDB) GetTransientState(addr common.Address, key common.Hash) common.Hash {
-	panic("not supported by Polaris")
+	return sdb.transientStorage.Get(addr, key)
 }
 
-// TODO: `SetTransientState` implements the `PolarisStateDB` interface by setting the transient state
+// `GetTransientState` implements `stateDB`
 func (sdb *stateDB) SetTransientState(addr common.Address, key, value common.Hash) {
-	panic("not supported by Polaris")
+	prev := sdb.GetTransientState(addr, key)
```

Comment:
Might be better to just always set? Optimize for the more frequent case (writing to empty slot)